From aef9f1c06a6a4633c29a27e916fa2d313b4f8ec0 Mon Sep 17 00:00:00 2001
From: Dan McDonald <danmcd@joyent.com>
Date: Wed, 16 Jan 2019 11:42:24 -0500
Subject: [PATCH] OS-7516 so_krecv_unblock() double-mutex-exits Reviewed by:
 Patrick Mooney <patrick.mooney@joyent.com> Reviewed by: Jason King
 <jbk@joyent.com>

---
 usr/src/uts/common/fs/sockfs/sockcommon_subr.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/usr/src/uts/common/fs/sockfs/sockcommon_subr.c b/usr/src/uts/common/fs/sockfs/sockcommon_subr.c
index 7bdd64393b..df159a122c 100644
--- a/usr/src/uts/common/fs/sockfs/sockcommon_subr.c
+++ b/usr/src/uts/common/fs/sockfs/sockcommon_subr.c
@@ -24,7 +24,7 @@
  */
 /*
  * Copyright 2014, OmniTI Computer Consulting, Inc. All rights reserved.
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 #include <sys/types.h>
@@ -2517,5 +2517,8 @@ so_krecv_unblock(sonode_t *so)
 
 	so->so_rcv_queued = 0;
 	(void) so_check_flow_control(so);
-	mutex_exit(&so->so_lock);
+	/*
+	 * so_check_flow_control() always drops so->so_lock, so we won't
+	 * need to drop it ourselves.
+	 */
 }
-- 
2.21.0

